appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/eesecurity/SignedJWTIdentityStore.java

```diff
@@ -179,141 +151,14 @@ public CredentialValidationResult validate(SignedJWTCredential signedJWTCredenti
-        return readPublicKeyFromLocation("/publicKey.pem", null);
-    }
-
-    private Optional<PublicKey> readMPEmbeddedPublicKey(String keyID) throws Exception {
-        Optional<String> key = config.getOptionalValue(VERIFIER_PUBLIC_KEY, String.class);
-        if (!key.isPresent()) {
-            return Optional.empty();
-        }
-        return createPublicKey(key.get(), keyID);
-    }
-
-    private Optional<PublicKey> readMPPublicKeyFromLocation(String keyID) throws Exception {
-        Optional<String> locationOpt = config.getOptionalValue(VERIFIER_PUBLIC_KEY_LOCATION, String.class);
-
-        if (!locationOpt.isPresent()) {
-            return Optional.empty();
-        }
-
-        String publicKeyLocation = locationOpt.get();
-
-        return readPublicKeyFromLocation(publicKeyLocation, keyID);
-    }
-
-    private Optional<PublicKey> readPublicKeyFromLocation(String publicKeyLocation, String keyID) throws Exception {
-
-        URL publicKeyURL = currentThread().getContextClassLoader().getResource(publicKeyLocation);
-
-        if (publicKeyURL == null) {
-            try {
-                publicKeyURL = new URL(publicKeyLocation);
-            } catch (MalformedURLException ex) {
-                publicKeyURL = null;
-            }
-        }
-        if (publicKeyURL == null) {
-            return Optional.empty();
-        }
-
-        URLConnection urlConnection = publicKeyURL.openConnection();
-        Charset charset = Charset.defaultCharset();
-        ContentType contentType = ContentType.newContentType(urlConnection.getContentType());
-        if(contentType != null) {
-            String charEncoding = contentType.getCharacterEncoding();
-            if(charEncoding != null) {
-                try {
-                    if (!Charset.isSupported(charEncoding)) {
-                        LOGGER.warning("Charset " + charEncoding + " for remote public key not supported, using default charset instead");
-                    } else {
-                        charset = Charset.forName(contentType.getCharacterEncoding());
-                    }
-                }catch (IllegalCharsetNameException ex){
-                    LOGGER.severe("Charset " + ex.getCharsetName() + " for remote public key not support, Cause: " + ex.getMessage());
-                }
-            }
-        }
-        try (InputStream inputStream = urlConnection.getInputStream();
-             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream, charset))){
-            String keyContents = reader.lines().collect(Collectors.joining(System.lineSeparator()));
-            return createPublicKey(keyContents, keyID);
-        }
+    
+    private Duration readPublicKeyCacheTTL(Optional<Properties> properties) {
+        return properties
+        		.map(props -> props.getProperty("publicKey.cache.ttl"))
+        		.map(Long::valueOf)
+        		.map(Duration::ofMillis)
+        		.orElseGet( () -> Duration.ofMinutes(5));
```

Comment:
@fturizo To fix the time unit discrepancies, I change the type from `long` to `java.time.Duration`. 

I left the unit of `publicKey.cache.ttl` as milliseconds but I could see an argument for using seconds instead. Let me know if you have any thoughts on that.

Reply:
As long as it is consistent, you can keep the unit in milliseconds.